### PR TITLE
[Product description AI Tooltip] Move display conditions to use case

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Part of: #9975

## Description
Moves the tooltip display check conditions to use case and add unit tests.

## Testing instructions
Prerequisites
1. A store hosted on WPCOM
1. Logout of the app and login again to reset tooltip display counter. (Need this step to reset user defaults. Only applicable if you have already viewed the AI tooltip 3 times)

**Steps**
1. Install and login into the store. 
2. Navigate to the Products tab
3. Open a product with an empty product description
4. Validate that you can see the tooltip below the "Write with AI" button
5. Open a product with a non-empty description
6. Validate that you don't see a tooltip below the "Write with AI" button
7. Turn off `productDescriptionAIFromStoreOnboarding` feature flag and repeat steps 1-3
8. Validate that the tooltip is not displayed. Additionally, observe that the "Write with AI" button will not be visible.

- [x] @selanthiraiyan Test that the tooltip is not displayed for self-hosted stores

**Edge case fixed in this PR**

While testing, I noticed an issue where the tooltip is presented when not needed (while rotating the device) 

- Have an empty description and open product to present the tooltip
- Add description
- Navigate back and validate tooltip is not shown
- The tooltip appears upon rotation even when the description is empty

The above edge case is fixed in https://github.com/woocommerce/woocommerce-ios/commit/dbb99bab9f159c0647669c864008c2877cdbbe39 and https://github.com/woocommerce/woocommerce-ios/commit/e725d93eb90d42f010939b3b7a7cd572df422176

## Screenshots

Product description
| Product description empty | Product description non-empty |
|--------|--------|
| EmptyDescription | NonEmptyDescription |

Feature flag
| `productDescriptionAIFromStoreOnboarding` true | `productDescriptionAIFromStoreOnboarding` false |
|--------|--------|
| FFOn | FFOff |

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/WooCommerceTests/ViewRelated/Feature Highlight/ProductDescriptionAITooltipUseCaseTests.swift [hunk 1]
```diff
@@ -1,41 +1,108 @@
 import XCTest
+import Yosemite
 @testable import WooCommerce
 
 final class ProductDescriptionAITooltipUseCaseTests: XCTestCase {
+    // MARK: Feature flag
```
Comment (jaclync) [inline WooCommerce/WooCommerceTests/ViewRelated/Feature Highlight/ProductDescriptionAITooltipUseCaseTests.swift:34]: thanks for adding the test cases!

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift [hunk 2]
```diff
@@ -72,10 +72,10 @@ final class ProductFormViewController<ViewModel: ProductFormViewModelProtocol>:
     private let aiEligibilityChecker: ProductFormAIEligibilityChecker
     private var descriptionAICoordinator: ProductDescriptionAICoordinator?
 
-    private lazy var tooltipUseCase = ProductDescriptionAITooltipUseCase()
+    private var tooltipUseCase: ProductDescriptionAITooltipUseCase?
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift]: ❓ any reasons for changing the lazy var to an optional variable? I thought it was nice initializing it only when needed and it's never reset

**Review by jaclync — APPROVED** [review]
Works as expected, thanks for the updates with additional test cases!

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift (hunk 2 above)
Comment (selanthiraiyan) [reply WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift]: Good point. Done in https://github.com/woocommerce/woocommerce-ios/pull/10073/commits/13cfd2a6d7f10b998af8455bb6f7677a2c4f5ea6

**Comment by selanthiraiyan** [comment]
Thanks for the review, @jaclync!

I have pushed the necessary changes. 

While testing, I noticed an issue where the tooltip is presented when not needed (while rotating the device) 

- Have an empty description and open product to present the tooltip
- Add description
- Navigate back and validate tooltip is not shown
- The tooltip appears upon rotation even when the description is empty

I fixed this issue about the `tooltipPresenter` not being deallocated properly in https://github.com/woocommerce/woocommerce-ios/pull/10073/commits/dbb99bab9f159c0647669c864008c2877cdbbe39 and https://github.com/woocommerce/woocommerce-ios/pull/10073/commits/e725d93eb90d42f010939b3b7a7cd572df422176

Could you kindly take another look at the PR?

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift [hunk 3]
```diff
@@ -614,24 +615,16 @@ private extension ProductFormViewController {
...
-
-        if let tooltip = tooltipPresenter?.tooltip {
-            tooltip.removeFromSuperview()
+        if let tooltipPresenter {
+            tooltipPresenter.removeTooltip()
             self.tooltipPresenter = nil
         }
 
-        guard product.description?.isEmpty == true else {
+        guard aiEligibilityChecker.isFeatureEnabled(.description) else {
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift]: ❓ I might have missed this earlier, does it make sense for this condition to be included in the use case's `shouldShowTooltip`?

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift [hunk 4]
```diff
@@ -614,24 +615,16 @@ private extension ProductFormViewController {
     }
 
     func configureTooltipPresenter() {
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift]: nit: we usually use `configure*` functions for one-time setup, while this can be called resetting `tooltipPresenter` more than once. maybe `update*` can be more clear?
```suggestion
    func updateTooltipPresenter() {
```

File: WooCommerce/Classes/ViewRelated/Feature Highlight/TooltipPresenter.swift [hunk 5]
```diff
@@ -117,6 +117,13 @@ final class TooltipPresenter {
         }
     }
 
+    // Silently removes the tooltip without firing the `primaryTooltipAction` callback
+    //
+    func removeTooltip() {
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Feature Highlight/TooltipPresenter.swift:121]: nit: the code seems to be the same as L115-116. Maybe this new function can be called there as well?

non-blocking future suggestion: since this is now a public function, it can also be unit tested about the notification at least.

**Review by jaclync — APPROVED** [review]
The updates LGTM 👍 nice catch on this scenario.

> While testing, I noticed an issue where the tooltip is presented when not needed (while rotating the device)
> 
> * Have an empty description and open product to present the tooltip
> * Add description
> * Navigate back and validate tooltip is not shown
> * The tooltip appears upon rotation even when the description is empty

Maybe this can be included in the testing steps in the PR description for future reference?

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift (hunk 3 above)
Comment (selanthiraiyan) [reply WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift]: Done in https://github.com/woocommerce/woocommerce-ios/pull/10073/commits/5ced25e6bd2951a0971829912cbb95b55c6b8058

I had this earlier and reverted the change to avoid initializing when `ProductDescriptionAITooltipUseCase` when AI is not eligible. I missed adding it back when I made the variable lazy. 👍

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift (hunk 4 above)
Comment (selanthiraiyan) [reply WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift]: Done in e858ff7a7b1eb24c11464dc3e75cbd08e09277fb

File: WooCommerce/Classes/ViewRelated/Feature Highlight/TooltipPresenter.swift (hunk 5 above)
Comment (selanthiraiyan) [reply WooCommerce/Classes/ViewRelated/Feature Highlight/TooltipPresenter.swift:121]: > the code seems to be the same as L115-116. Maybe this new function can be called there as well?

Done in 0bbb98f3c5808f6aee4c83523300d015a77932a9

Comment (selanthiraiyan) [reply WooCommerce/Classes/ViewRelated/Feature Highlight/TooltipPresenter.swift:121]: > non-blocking future suggestion: since this is now a public function, it can also be unit tested about the notification at least.

Done in 00d75c96a09c2bb72a8ed7e44211b286c5711808
